### ISSUE TITLE
UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`

### ISSUE THREAD

**Issue by Angryrou** [Tony-Y/pytorch_warmup#13]
Hi Tony, I got a similar warning as #5 on using `warmup.UntunedLinearWarmup` after I upgraded my Pytorch to 1.12.1 

_UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate_

Would you help double-check whether I can still ignore the warning in my pytorch version?

**Comment by Tony-Y** [comment]
I didn't get the warning when running [sample code](https://gist.github.com/Tony-Y/a15fe78094aba6a77555cf40c52e4521) for `pytorch_lightning` using `PyTorch 1.12.1`.  (This code was made due to #8) Your problem is probably caused by a misuse. Could you try [the official example](https://github.com/Tony-Y/pytorch_warmup/blob/master/examples/emnist/main.py) in your PyTorch environment?

**Comment by Angryrou** [comment]
Thanks! After refactoring my code to align with the official example, my problem has been solved. You are perfectly right.
